python/tests/kernel/test_kernel_features.py

```diff
@@ -1617,6 +1617,102 @@ def kd():
     counts = cudaq.sample(kd)
     assert len(counts) == 2 and '0' in counts and '1' in counts 
     
+def test_custom_kernel_type():
```

Comment:
Can you add one representative test to 'remote' backend as well?

Reply:
Does the C++ frontend support custom user structs for remote execution?

Reply:
```
QuakeSynthesizer.cpp:608
// The struct type ends up as a i64 in the thunk kernel args pointer, so
// just skip ahead. TODO: add support for struct types!
```

Reply:
This should be a follow-on work. See issue #1987